### ISSUE TITLE
"only one socket address is normally permitted"

### ISSUE THREAD

**Issue by islaitala** [RiptideNetworking/Riptide#82]
Yes, I submitted the same issue before.

I don't know how to reproduce, I get the issue after some time when I have stopped and started play mode in unity editor multiple times. It gets fixed when I restart the unity editor.

**Comment by tom-weiland** [comment]
#53 was the previous issue (just for reference).

I had someone else complain about this same error [on Discord](https://discord.com/channels/409147571928563712/1064008776102969364/1064008776102969364) just the other day. It's caused by the socket not being shut down correctly (even though—as far as I know—it should be when calling `Server.Stop` in `OnApplicationQuit`), and restarting the editor fixes it because that cleans it all up.

I myself haven't run into this error in several months, which makes debugging it virtually impossible. Any additional information about in which scenarios/what you were doing when the error occurs would be extremely helpful, particularly if you notice any patterns.

Even just putting a `Debug.Log` line after the `Server.Stop` call to confirm that it's running when you get the error would be a good start.

**Comment by tom-weiland** [comment]
That's helpful to know, but still isn't definitive—either Mirror has the same problem or something about the Unity editor prevents it from shutting sockets down properly...

The last time I personally ran into this (months and months ago), I think it may have been after making code changes while the editor was in play mode. Perhaps when Unity rebuilds/reloads the code it either messes with running sockets or prevents the shutdown code from running? That was a long time ago and may have just been a one-off though, so if anyone else can provide more context regarding the conditions/timing/circumstances of this error, that'd be a big help.

**Comment by megalithos** [comment]
Yeah that sounds like a valid theory.
My guess is that the bug happens with some chance (like 0,1% chance) (lol) when the code is edited & unity is stopped while it is running like you said. So it would be internal bug in unity itself.

**Comment by tom-weiland** [comment]
> when the code is edited & unity is stopped while it is running

Yeah, I've noticed it happening under those circumstances, but I can't confirm that that's the *only* time it happens :/

**Comment by hedgewizards** [comment]
a windows workaround slightly better than restarting unity:

- open command prompt as administrator
- run the command `netsh winsock reset`

this resets ALL your sockets so it might break other applications running on your computer.
